src/stats.jl

```diff
@@ -57,36 +57,41 @@ All other variables can be input as keyword arguments.
 
 Notice that `GenericExecutionStats` does not compute anything, it simply stores.
 """
-mutable struct GenericExecutionStats <: AbstractExecutionStats
+mutable struct GenericExecutionStats{T, V} <: AbstractExecutionStats
   status::Symbol
-  solution::Vector # x
-  objective::Real # f(x)
-  dual_feas::Real # ‖∇f(x)‖₂ for unc, ‖P[x - ∇f(x)] - x‖₂ for bnd, etc.
-  primal_feas::Real # ‖c(x)‖ for equalities
-  multipliers::Vector # y
-  multipliers_L::Vector # zL
-  multipliers_U::Vector # zU
+  solution::V # x
+  objective::T # f(x)
+  dual_feas::T # ‖∇f(x)‖₂ for unc, ‖P[x - ∇f(x)] - x‖₂ for bnd, etc.
+  primal_feas::T # ‖c(x)‖ for equalities
+  multipliers # y
+  multipliers_L # zL
+  multipliers_U # zU
   iter::Int
   counters::NLPModels.NLSCounters
   elapsed_time::Real
   solver_specific::Dict{Symbol, Any}
 end
 
-function GenericExecutionStats(
+function GenericExecutionStats(status::Symbol, nlp::AbstractNLPModel; kwargs...)
+	T = eltype(nlp.meta.x0)
+	V = typeof(nlp.meta.x0)
+	return GenericExecutionStats{T, V}(status, nlp; kwargs...)
+end
```

Comment:
```suggestion
function GenericExecutionStats(
  status::Symbol,
  nlp::AbstractNLPModel;
  solution::V = eltype(nlp.meta.x0)[],
  kwargs...
) where V
	T = eltype(solution)
	return GenericExecutionStats{T, V}(status, nlp; solution=solution, kwargs...)
end
```
It needed more information. I tested with [this](https://gist.github.com/abelsiqueira/1597884bd2c9f52369b41b18fd222865). Unfortunately, I don't have access to CUArrays to test with different `V`s.

Reply:
Than you! I just tested your example in a google collab with CUDA and it worked

```
using SolverCore, NLPModels, CUDA

mutable struct DummyModel <: AbstractNLPModel
  meta :: AbstractNLPModelMeta
  counters :: Counters
end

DummyModel() = DummyModel(NLPModelMeta(2, x0=zeros(2)), Counters())

#%%
nlp = DummyModel()

output = GenericExecutionStats(:unknown, nlp)
print(output)

#%%
x = CUDA.ones(Float64, 2)
output = GenericExecutionStats(:unknown, nlp, solution=x)
print(output)

#%%
output = GenericExecutionStats(:unknown, nlp, solution=x, dual_feas=one(Float64))
print(output)
```